File: ungs-docsys-ms/src/main/java/com/ungs/docsys/dtos/TechnicalSkillRequestDto.java
package com.ungs.docsys.dtos;

import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.Size;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class TechnicalSkillRequestDto {
    @NotBlank(message = "Name must not be blank")
    @Size(max = 100, message = "Name must not exceed 100 characters")
    private String name;
    @NotBlank(message = "Level must not be blank")
    @Size(max = 50, message = "Level must not exceed 50 characters")
    private String level;
}

File: ungs-docsys-bff/src/modules/resume-user/dtos/technical-skill-request.dto.ts
import {
  IsOptional,
  IsString,
  MaxLength
} from 'class-validator';

export class TechnicalSkillRequestDto {
  @IsOptional()
  @IsString()
  @MaxLength(100, { message: 'Name must not exceed 100 characters' })
  name?: string;

  @IsOptional()
  @IsString()
  @MaxLength(50, { message: 'Level must not exceed 50 characters' })
  level?: string;
}